core/src/main/java/org/apache/iceberg/view/ViewVersionReplace.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import static org.apache.iceberg.TableProperties.COMMIT_MAX_RETRY_WAIT_MS;
+import static org.apache.iceberg.TableProperties.COMMIT_MAX_RETRY_WAIT_MS_DEFAULT;
+import static org.apache.iceberg.TableProperties.COMMIT_MIN_RETRY_WAIT_MS;
+import static org.apache.iceberg.TableProperties.COMMIT_MIN_RETRY_WAIT_MS_DEFAULT;
+import static org.apache.iceberg.TableProperties.COMMIT_NUM_RETRIES;
+import static org.apache.iceberg.TableProperties.COMMIT_NUM_RETRIES_DEFAULT;
+import static org.apache.iceberg.TableProperties.COMMIT_TOTAL_RETRY_TIME_MS;
+import static org.apache.iceberg.TableProperties.COMMIT_TOTAL_RETRY_TIME_MS_DEFAULT;
+
+import java.util.List;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.exceptions.CommitFailedException;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.util.PropertyUtil;
+import org.apache.iceberg.util.Tasks;
+
+public class ViewVersionReplace implements ReplaceViewVersion {
+  private final ViewOperations ops;
+  private final List<ViewRepresentation> viewRepresentationsToAdd = Lists.newArrayList();
+  private ViewMetadata base;
+  private Namespace defaultNamespace;
+  private String defaultCatalog;
+  private Schema schema;
+
+  ViewVersionReplace(ViewOperations ops) {
+    this.ops = ops;
+    this.base = ops.current();
+  }
+
+  @Override
+  public ViewVersion apply() {
+    this.base = ops.refresh();
+
+    ViewVersion viewVersion = base.currentVersion();
+
+    if (!viewRepresentationsToAdd.isEmpty()) {
+      long timestampMillis = System.currentTimeMillis();
+      int newVersion = viewVersion.versionId() + 1;
+
+      return ImmutableViewVersion.builder()
+          .versionId(newVersion)
+          .timestampMillis(timestampMillis)
+          .schemaId(null != schema ? schema.schemaId() : viewVersion.schemaId())
+          .defaultNamespace(defaultNamespace)
+          .defaultCatalog(defaultCatalog)
+          .putSummary("operation", "replace")
+          .addAllRepresentations(viewRepresentationsToAdd)
+          .build();
+    }
+
+    return viewVersion;
```

Comment:
I think this is incorrect. This seems to imply that if there are no representations added that the replace is a noop, but that's clearly not the caller's intent. I think that this API should fail if there were no representations added.

Reply:
I've updated this so that empty commits are not allowed